stormevents/nhc/track.py

```diff
@@ -1196,3 +1151,83 @@ def separate_tracks(data: DataFrame) -> Dict[str, Dict[str, DataFrame]]:
             ] = track_data
 
     return tracks
+
+
+def combine_tracks(tracks: Dict[str, Dict[str, DataFrame]]) -> DataFrame:
+    """
+    combine tracks separated using `separate_tracks`
+
+    :param tracks: dictionary of forecasts for each advisory (aside from best track ``BEST``, which only has one hindcast)
+    :return: data frame of track
+    """
+
+    return pandas.concat([df for adv_trk in tracks.values() for df in adv_trk.values()])
+
+
+def correct_ofcl_based_on_carq_n_hollandb(
+    tracks: Dict[str, Dict[str, DataFrame]]
+) -> Dict[str, Dict[str, DataFrame]]:
+    """
+    Correct official forecast using consensus track along with holland-b
+    relation
+
+    :param tracks: dictionary of forecasts for each advisory (aside from best track ``BEST``, which only has one hindcast)
+    :return: dictionary of forecasts for each advisory (aside from best track ``BEST``, which only has one hindcast) with corrected OFCL
+    """
+
+    ofcl_tracks = tracks["OFCL"]
+    carq_tracks = tracks["CARQ"]
+
+    corr_ofcl_tracks = dict()
+
+    for initial_time, forecast in ofcl_tracks.items():
+        if initial_time in carq_tracks:
+            carq_forecast = carq_tracks[initial_time]
+        else:
+            carq_forecast = carq_tracks[list(carq_tracks)[0]]
+
+        relation = HollandBRelation()
+        holland_b = relation.holland_b(
+            max_sustained_wind_speed=carq_forecast["max_sustained_wind_speed"],
+            background_pressure=carq_forecast["background_pressure"],
+            central_pressure=carq_forecast["central_pressure"],
+        )
+
+        holland_b[holland_b == numpy.inf] = numpy.nan
+        holland_b = numpy.nanmean(holland_b)
+
+        # Get CARQ from forecast hour 0 and isotach 34kt (i.e. the first item)
+        carq_ref = carq_forecast.loc[carq_forecast.forecast_hours == 0].iloc[0]
```

Comment:
`CARQ` track entry to be used is now the first entry of hour-0 forecast (i.e. `34 kt`) #65